### PR TITLE
add cli commons to factor out common parsing code

### PR DESCRIPTION
**Pull request by cgardens**
## What
* With Cloud we are leaning towards writing more scripts in java. We currently use apache.commons.cli for this. While I recommend in this [issue](https://github.com/airbytehq/airbyte-internal-issues/issues/300) we move away from it (and use picocli instead), this PR adds a couple quick helpers to factor out common logic we use for parsing command line args with the current tool. It also helps concentrate all of the use of the cli tool, so it'll be easier to excise. 
* Once it is released, I'll switch cloud to depend on it.

### CODE REVIEW

**Review by sherifnada — APPROVED** [review]
unit tests and a question but otherwise LGTM

**Review by benmoriceau — APPROVED** [review]
LGTM, just a small comment on the use of an enum.

File: airbyte-commons-cli/build.gradle [hunk 1]
```diff
@@ -0,0 +1,7 @@
+plugins {
```
[the remark this answers is not in the document]
Comment (cgardens) [reply airbyte-commons-cli/build.gradle:1]: lol. i guess if you have to ask it's not overkill :D I'll add it.

File: airbyte-commons-cli/src/test/java/io/airbyte/commons/cli/ClisTest.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+import org.junit.jupiter.api.Test;
+
+class ClisTest {
+
+  @Test
+  void testGetTaggedImageName() {
+    final String repository = "airbyte/repo";
+    final String tag = "12.3";
+    assertEquals("airbyte/repo:12.3", Clis.getTaggedImageName(repository, tag));
```
[the remark this answers is not in the document]
Comment (cgardens) [reply airbyte-commons-cli/src/test/java/io/airbyte/commons/cli/ClisTest.java:17]: not thick. i fix.

File: airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationCliParser.java [hunk 3]
```diff
@@ -87,26 +70,46 @@ private static IntegrationConfig parseOptions(final String[] args, final Command
...
+            .longOpt(JavaBaseConstants.ARGS_CATALOG_KEY)
+            .desc(JavaBaseConstants.ARGS_CATALOG_DESC)
+            .hasArg(true)
+            .build());
       }
       default -> throw new IllegalStateException("Unexpected value: " + command);
     }
 
-    final CommandLine parsed = runParse(options, args, command);
+    final CommandLine parsed = Clis.parse(args, options, command.toString().toLowerCase());
```
[the remark this answers is not in the document]
Comment (cgardens) [reply airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationCliParser.java:112]: it's the trade off of DRYing this is that the common `parse` function can't really reasonably know about the enum, right?

**Pull request merged**